### ISSUE TITLE
Tracking: fluid typography 

### ISSUE THREAD

**Issue by ramonjd** [WordPress/gutenberg#44888]
An overview issue to track **fluid typography** follow ups and enhancements.

Related:

- https://github.com/WordPress/gutenberg/issues/34345
- https://github.com/WordPress/gutenberg/issues/34641

## Fluid typography base functionality 
- https://github.com/WordPress/gutenberg/issues/24480
    - https://github.com/WordPress/gutenberg/pull/39529
    - https://github.com/WordPress/gutenberg/pull/42688
    - https://github.com/WordPress/gutenberg/pull/42757
- https://github.com/WordPress/gutenberg/issues/44758
    - https://github.com/WordPress/gutenberg/pull/44762
    - https://github.com/WordPress/gutenberg/pull/44761
    - https://github.com/WordPress/gutenberg/pull/44764
    - https://github.com/WordPress/gutenberg/pull/44765
    - https://github.com/WordPress/gutenberg/pull/44807
    - https://github.com/WordPress/gutenberg/pull/44847
    - https://github.com/WordPress/gutenberg/pull/44852 
- https://github.com/WordPress/gutenberg/issues/44957
    - https://github.com/WordPress/gutenberg/pull/44993    
- https://github.com/WordPress/gutenberg/issues/45504
    - https://github.com/WordPress/gutenberg/pull/45536 
- https://github.com/WordPress/gutenberg/pull/42489
- https://github.com/WordPress/gutenberg/pull/44967
- https://github.com/WordPress/gutenberg/pull/49707
- https://github.com/WordPress/gutenberg/pull/49815
    - https://github.com/WordPress/gutenberg/pull/51146
    - https://github.com/WordPress/gutenberg/pull/51188
    - https://github.com/WordPress/gutenberg/pull/51516
- https://github.com/WordPress/gutenberg/pull/53081
    - https://github.com/WordPress/gutenberg/pull/53082 
- https://github.com/WordPress/gutenberg/pull/53551
    - https://github.com/WordPress/gutenberg/pull/53554
- https://github.com/WordPress/gutenberg/pull/57866
 
## Current improvements, fixes and extensions
- https://github.com/WordPress/gutenberg/issues/58135
    - https://github.com/WordPress/gutenberg/pull/58362
- https://github.com/WordPress/gutenberg/issues/53525
- https://github.com/WordPress/gutenberg/issues/57607
- Regression: clamp calculations are broken in the editor since https://github.com/WordPress/gutenberg/pull/56912#discussion_r1452946735

## Wishlist
- Add further config values. See [comment](https://github.com/WordPress/gutenberg/pull/49707#pullrequestreview-1389308024) and [previous implementation](https://github.com/WordPress/gutenberg/pull/42489#issuecomment-1347672614). Or more control through filters. Max and min viewport have been already implemented in https://github.com/WordPress/gutenberg/pull/53081

### Bugs, tasks, and issues not covered here
For bug reports and smaller tasks, we can use the [[Feature] Typography](https://github.com/WordPress/gutenberg/issues?q=label%3A%22%5BFeature%5D+Typography%22+fluid+) issue label for tracking and visibility.

## Closed unfixed
- https://github.com/WordPress/gutenberg/issues/45502

**Comment by eric-michel** [comment]
@ramonjd I have a quick question regarding https://github.com/WordPress/gutenberg/pull/49815.

Is there a way to independently specify the maximum viewport width for fluid font size calculations? The change in 49815 will significantly reduce the width at which our font sizes will begin shrinking, and could break certain layouts on legacy sites. Going forward, I definitely support using `settings.layout.wideSize` as the maximum viewport width, but I would like to keep `1600px` as the max value for existing sites just to avoid any possible issues.

If this is something I can set in `theme.json` right now on those older sites in anticipation of this PR becoming active in WP core, that would be super helpful.

edit: Looking at the `theme.json` schema, I don't see a setting for this. I think it would be very useful to make both values (the max and the min) for the calculation independently customizable for those who wish to do so.

**Comment by ramonjd** [comment]
Thanks for the question @eric-michel 

This is something I wanted to do much earlier on. See [comment](https://github.com/WordPress/gutenberg/pull/42489#issuecomment-1347672614) and [commit history](https://github.com/WordPress/gutenberg/pull/42489/commits/642b3c70b49174d09425477cd6039e91a3e3934b#diff-65c0d061536e5fa3369aba19faf1f475337c549f26828424fa224908f0bcf137) of:

- https://github.com/WordPress/gutenberg/pull/42489

You're right, it's not possible to configure at the moment, unless you fiddle with the layout width.

I think it should be possible for theme authors to configure both the min and the max in theme.json. I've added a PR here to remind myself to come back to it 😄 

- https://github.com/WordPress/gutenberg/pull/53081

**Comment by porg** [comment]
Please integrate #53525 to section "Wishlist"

**Comment by briceduclos** [comment]
@ramonjd Are there any plans to add the minimum font size factor as a configurable option in theme.json?

```
$minimum_font_size_factor     = min( max( 1 - 0.075 * log( $preferred_font_size_in_px, 2 ), $default_minimum_font_size_factor_min ), $default_minimum_font_size_factor_max );
$calculated_minimum_font_size = round( $preferred_size['value'] * $minimum_font_size_factor, 3 );

```
The min and max viewport width configurable options are a great improvement. Adding the min size factor would offer precise control over the responsiveness of font sizes. For my website, the current factor makes the curve move towards the minimum too slowly at the beginning and too quickly at the end.

**Comment by ramonjd** [comment]
> Are there any plans to add the minimum font size factor as a configurable option in theme.json?

Thanks for the ping. 

Allowing more config options was something that I explored in a [previous, now closed, PR.](https://github.com/WordPress/gutenberg/pull/42489#issuecomment-1347672614). 

`scaleFactor` is definitely something that may be useful to tweak as it controls how fast the font sizes change between min and max limits.

Another idea, and something I'd like to pursue, is leveraging filters to override some things. 

It was decided to move slowly with this sort of stuff to provide room for more testing as every change as the potential to affect existing sites implementing the fluid font size system.

It's on the wishlist in the description of this tracking issue 👍🏻

**Comment by annezazu** [comment]
Wanted to note adding this to the mix: https://github.com/WordPress/gutenberg/issues/58135

**Comment by ramonjd** [comment]
I'm going to close this tracking PR, as the main feature has been implemented since WordPress 6.1.

Subsequent bugfixes and stability updates are complete.

The outstanding issues can stand on their own. 

- https://github.com/WordPress/gutenberg/pull/58362 is close to completion
- https://github.com/WordPress/gutenberg/issues/53525 might be feasible if the above PR merges
- https://github.com/WordPress/gutenberg/issues/57607 is a wider PHP issue and not isolated to fluid typography
